crates/api_common/src/utils.rs

```diff
@@ -754,15 +724,6 @@ pub async fn remove_or_restore_user_data(
 
   // Only these actions are possible when removing, not restoring
   if removed {
-    // Purge user images
-    let person = Person::read(pool, banned_person_id).await?;
-    if let Some(avatar) = person.avatar {
-      purge_image_from_pictrs(&avatar, context).await.ok();
-    }
-    if let Some(banner) = person.banner {
-      purge_image_from_pictrs(&banner, context).await.ok();
-    }
-
```

Comment:
Here's where the actual bug was.

In place of this block, we need to add `delete_local_user_images(person_id, context).await.ok();`

That way we can ensure that when *removing* a user's data, it will only remove the local images for that person. 

Currently, the `delete_local_user_images` function is only used for purge, but it needs to also be used when banning + removing content.

Reply:
if media deletion is desired separate from purging, imo this should be a separate option to pass when banning a user, as banning with removal is generally something that should be reversible.

Reply:
do you want me to add a new optional api field `Option<bool> remove_media`, defaulting to `false`, within this PR?

Reply:
i think it would make more sense to keep that for a separate pr, in which the federation aspect can also be implemented.

Reply:
updated to delete all media uploaded by the user on bans with content removal now.
seems that other people (matrix) also expect media deletion, so even if this would get changed it is likely out of the scope of this PR.